Sources/Dependencies/ConcurrencySupport/MainSerialExecutor.swift

```diff
@@ -0,0 +1,30 @@
+import Foundation
+
+@_spi(Concurrency) public func withMainSerialExecutor<T>(
+  @_implicitSelfCapture operation: () async throws -> T
+) async rethrows -> T {
+  guard let pointer = swift_task_enqueueGlobal_hook else { return try await operation() }
+  let hook = pointer.pointee
+  defer { pointer.pointee = hook }
+  pointer.pointee = { job, original in
+    MainActor.shared.enqueue(job)
+  }
+  return try await operation()
+}
+
+// here be dragons
+private typealias Orig = @convention(thin) (UnownedJob) -> Void
+private typealias Hook = @convention(thin) (UnownedJob, Orig) -> Void
+private var swift_task_enqueueGlobal_hook: UnsafeMutablePointer<Hook>? = {
+  var info = Dl_info()
+  guard
+    withUnsafePointer(to: TaskPriority.self, {
```

Comment:
using this to obtain a handle to `libswift_Concurrency.{dylib,so}` without hardcoding a path — open to suggestions on better ways to do this. Ideally the image path would be resolved at launch time by the linker (which can be accelerated by dyld closures etc) but since this API should really only be used in tests I'm begrudgingly okay with the `dladdr` approach.